modules/auxiliary/gather/http/cve_2021_27850_apache_tapestry_hmac_key.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Apache Tapestry HMAC secret key leak',
+        'Description' => %q{
+          This exploit finds the HMAC secret key used in Java serizalization by Apache Tapestry. This key
+          is located in the file AppModule.class by default and looks like the standard representation of UUID in hex digits (hd) :
+          6hd-4hd-4hd-4hd-12hd
+          If the HMAC key has been changed to look differently, this module won't find the key because it tries to download the file
```

Comment:
Is there a general regex that can be used to capture any HMAC key?

Reply:
There could be one by looking more precisely in the binary file downloaded as the line to modify the HMAC key must look like this : `configuration.add(SymbolConstants.HMAC_PASSPHRASE, "your_hmac_key");`
So a better regex could be something that looks between `configuration.add(SymbolConstants.HMAC_PASSPHRASE,` and the last `)`. But the last time I tried I had some issues with some binary symbols between those two. So I decided to finish this module with a basic research first, and then maybe improve it in a second time.